### PR TITLE
EDGINREACH-4: Create edge-inn-reach folio module

### PR DESCRIPTION
**Pull request by oleksandr-yatsenko1**
EDGINREACH-4: Create edge-inn-reach folio module

## Purpose
[EDGINREACH-4](https://issues.folio.org/browse/EDGINREACH-4)
Create new Spring based EDGE module and pass it through standard new module creation procedure. 

## Approach
Number of order transactions have to be greater than 0
Add additional checking of amount of pending encumbrances to avoid transaction creating when there are no encumbrances at all. 

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+        <groupId>com.fasterxml.jackson</groupId>
+        <artifactId>jackson-bom</artifactId>
+        <type>pom</type>
+        <scope>import</scope>
+        <version>${jackson.version}</version>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+
+  <dependencies>
```
Comment (oleksandr-dekin) [inline pom.xml:62]: Please add actuator for health check, all modules which are follow spring way uses actuator
```
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-actuator</artifactId>
    </dependency>
```

File: descriptors/DeploymentDescriptor-template.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "srvcId": "${artifactId}-${version}",
```
Comment (oleksandr-dekin) [inline descriptors/DeploymentDescriptor-template.json:2]: Please change to "srvcId": "@artifactId@-@version@",

File: descriptors/ModuleDescriptor-template.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
+{
+  "id": "${artifactId}-${version}",
```
Comment (oleksandr-dekin) [inline descriptors/ModuleDescriptor-template.json:2]: The same

File: src/test/java/org/folio/edge/EdgeInnReachApplicationTests.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package org.folio.edge;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class EdgeInnReachApplicationTests {
```
Comment (oleksandr-dekin) [inline src/test/java/org/folio/edge/EdgeInnReachApplicationTests.java:7]: You can remove this test and exclude class EdgeInnReachApplication   in the pom.xml in the section `sonar.exclusions`

File: pom.xml (hunk 1 above)
Comment (oleksandr-yatsenko1) [reply pom.xml:62]: Done

File: descriptors/DeploymentDescriptor-template.json (hunk 2 above)
Comment (oleksandr-yatsenko1) [reply descriptors/DeploymentDescriptor-template.json:2]: Done

File: descriptors/ModuleDescriptor-template.json (hunk 3 above)
Comment (oleksandr-yatsenko1) [reply descriptors/ModuleDescriptor-template.json:2]: Done

File: src/test/java/org/folio/edge/EdgeInnReachApplicationTests.java (hunk 4 above)
Comment (oleksandr-yatsenko1) [reply src/test/java/org/folio/edge/EdgeInnReachApplicationTests.java:7]: If I understand correctly you meant just remove test for spring app start point and configure property name to skip sonar checking for start point of this module. Correct?

**Pull request merged**